app/scripts/controllers/rif/rns/register/index.js

```diff
@@ -25,63 +26,60 @@ export default class RnsRegister extends RnsJsDelegate {
+    const domainHash = web3Utils.sha3(cleanDomainName);
     const secret = generateRandomSecret();
     return new Promise((resolve, reject) => {
-      this.call(this.fifsContractInstance, 'makeCommitment', [domainHash, this.address, secret])
-        .then(commitment => {
-          console.debug('Commitment received', commitment);
-          this.sendTransaction(this.fifsContractInstance, 'commit', [commitment])
-            .then(result => {
-              console.debug('Commitment committed', result);
-              const state = this.getStoreState();
-              if (!state.register[this.address]) {
-                state.register[this.address] = {};
-              }
-              state.register[this.address].domainRegister = {
-                secret,
-                domainName,
-                yearsToRegister,
-              };
-              this.updateStoreState(state);
+      this.getDomainCost(cleanDomainName, yearsToRegister)
+        .then(rifCost => {
+          this.call(this.fifsAddrRegistrarInstance, 'makeCommitment', [domainHash, this.address, secret])
+            .then(commitment => {
+              console.debug('Commitment received', commitment);
+              this.send(this.fifsAddrRegistrarInstance, 'commit', [commitment])
+                .then(result => {
+                  console.debug('Commitment committed', result);
+                  const state = this.getStoreState();
+                  if (!state.register[this.address]) {
+                    state.register[this.address] = {
+                      domainRegister: [],
+                    };
+                  }
+                  state.register[this.address].domainRegister[cleanDomainName] = {
+                    secret,
+                    yearsToRegister,
+                    rifCost,
+                  };
+                  this.updateStoreState(state);
+                }).catch(error => reject(error));
+              resolve(commitment);
             }).catch(error => reject(error));
-          resolve(secret);
         }).catch(error => reject(error));
     });
   }
 
   /**
    * Calculates the rif cost for a domain
    * It uses the formula here: https://github.com/rnsdomains/rns-rskregistrar#name-price
-   * @param yearsToRegister
+   * @param domainName the domain name to ask for
+   * @param yearsToRegister the amount of years to ask
    */
-  getDomainCost (yearsToRegister) {
-    if (yearsToRegister && yearsToRegister > 0) {
-      if (yearsToRegister === 1) {
-        return 2;
-      } else if (yearsToRegister === 2) {
-        return 4;
-      } else {
-        const kMember = yearsToRegister - 2;
-        return 4 + kMember;
-      }
-    } else {
-      return -1;
-    }
+  getDomainCost (domainName, yearsToRegister) {
+    const cleanDomainName = this.cleanDomainFromRskPrefix(domainName);
+    return this.call(this.fifsAddrRegistrarInstance, 'price', [domainName, 0, yearsToRegister]);
```

Comment:
What is the purpose of sending 0 as expiration?

Reply:
It's how it was on the react manager code, i thought to not change that since it was working but maybe i should see why is working like that, i didn't want to spend time on that, but i can do it.

Reply:
Do you agree on research what is the purpose of that parameter, and move the 0 value to a constant for now?